plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -278,12 +299,31 @@ func (a *Elasticsearch) Write(metrics []telegraf.Metric) error {
 		// to send the metric to the correct time-based index
 		indexName := a.GetIndexName(a.IndexName, metric.Time(), a.TagKeys, metric.Tags())
 
+		// Handle NaN and inf field-values
+		fields := make(map[string]interface{})
+		for k, value := range metric.Fields() {
+			v, ok := value.(float64)
+			if !ok || a.FloatHandling == "none" || !(math.IsNaN(v) || math.IsInf(v, 0)) {
+				fields[k] = value
```

Comment:
Can you help ensure my understanding of this block is correct:  we save the value as-is, if any of 4 conditions are true:

* we failed to convert to float64
* float handling is "none"
* the value is **not** "not-a-number"
* the value is infinity

I understand doing this if float handling is "none", but I'm not sure I'm following the other cases.

Reply:
No the last tow condition `!(math.IsNaN(v) || math.IsInf(v, 0))` are wrapped in a bracket which is "it's neither a `NaN` nor an `Inf`" as if either `math.Is...` condition is `true` the whole expression gets `false`.

For the conditions:

1. We only want to fix `float64`s, so if we encounter any other type we simply keep the number. We might want to add `float32` but that's another topic.
2. You got that one.
3. & 4. See above. If it's neither `NaN` nor `Inf` we do nothing and keep the value.

Reply:
Ah I totally missed the extra bracket trying to read each or segment by itself. Thanks.